dataset/src/main/scala/frameless/TypedEncoder.scala

```diff
@@ -250,8 +251,9 @@ object TypedEncoder {
         }
     }
 
-  implicit def collectionEncoder[C[X] <: Seq[X], T]
+  implicit def collectionEncoder[C[_], T]
     (implicit
+      is: IsTraversableLike[C[T]] { type A = T },
```

Comment:
I think this constraint is unfortunately too generic. The implementation of `MapObjects` seems to work only on types which are assignable from `Seq`, `collection.Set` and `java.util.List` classes, but I'm not sure (and I can't see any guarantee) that these types are the same.

Reply:
:+1:, see [catalyst/ScalaReflection.scala#L324](https://github.com/apache/spark/blob/e07aee2165af4d301ae12005a6d9ffb030bc2650/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala#L324)